scipy/optimize/_gensa.py

```diff
@@ -0,0 +1,631 @@
+        probability of acceptance. The lower the acceptance parameter, the
+        smaller the probability of acceptance. It has to be any negative value.
+    maxcall : int, optional
+        Soft limit for the number of objective function calls. If the
+        algorithm is in the middle of a local search, this number will be
+        exceeded, the algorithm will stop just after the local search is
+        done.
+    args : tuple, optional
+        Any additional fixed parameters needed to
+        completely specify the objective function.
+    seed : int or `np.random.RandomState`, optional
+        If `seed` is not specified the `np.RandomState` singleton is used.
+        If `seed` is an int, a new `np.random.RandomState` instance is used,
+        seeded with seed.
+        If `seed` is already a `np.random.RandomState instance`, then that
+        `np.random.RandomState` instance is used.
+        Specify `seed` for repeatable minimizations. The random numbers
+        generated with this seed only affect the visiting distribution
+        function and new coordinates generation.
+
+    Returns
+    -------
+    res : OptimizeResult
+        The optimization result represented as a ``OptimizeResult`` object.
+        Important attributes are: ``x`` the solution array, ``fun`` the value
+        of the function at the solution, and ``message`` which describes the
+        cause of the termination.
+        See `OptimizeResult` for a description of other attributes.
+
+    Notes
+    -----
+    GenSA is an implementation of the General Simulated Annealing algorithm
+    (GSA [2]_). This stochastic approach generalizes CSA (Classical Simulated
+    Annealing) and FSA (Fast Simulated Annealing) to find the neighborhood of
+    minima, then calls a local method (lbfgsb) to find their exact value.
+    GenSA can process complicated and high dimension non-linear objective
+    functions with a large number of local minima as described by [6]_.
+
+    GSA uses a distorted Cauchy-Lorentz visiting distribution, with it shape
+    controlled by the parameter :math:`q_{v}`
+
+    .. math::
+
+        g_{q_{v}}(\\Delta x(t)) \\propto \\frac{ \\
+        \\left[T_{q_{v}}(t) \\right]^{-\\frac{D}{3-q_{v}}}}{ \\
+        \\left[{1+(q_{v}-1)\\frac{(\Delta x(t))^{2}} { \\
+        \\left[T_{q_{v}}(t)\\right]^{\\frac{2}{3-q_{v}}}}}\\right]^{ \\
+        \\frac{1}{q_{v}-1}+\\frac{D-1}{2}}}
+
+    Where :math:`t` is the artificial time. This visiting distribution is used
+    to generate a trial jump distance :math:`\Delta x(t)` of variable
+    :math:`x(t)` under artificial temperature :math:`T_{q_{v}}(t)`.
+
+    From the starting point, after calling the visiting distribution
+    function, the acceptance probability is computed as follows:
+
+    .. math::
+
+        p_{q_{a}} = \min{\{1,\\left[1-(1-q_{a}) \\beta \\Delta E \\right]^{ \\
+        \\frac{1}{1-q_{a}}}\\}}
+
+    Where :math:`q_{a}` is a parameter. For :math:`q_{a}<1`, zero acceptance
```

Comment:
Is `a` the `acceptparam`? It would be good if these equations are immediately relatable to the parameters used to call `gensa`. As an example look at [this equation](https://github.com/scipy/scipy/blob/master/scipy/optimize/_differentialevolution.py#L151). The `mutation` variable is the same as the mutation parameter used to call `differential_evolution`.